### DISCUSSION TITLE
how to set up gpu-id when training with single gpu based on version 1.1

### DISCUSSION THREAD

**Discussion by JiSuanJiDaWang** [open-mmlab/mmdetection3d#2019] [Q&A] [answered]
Hi, I used the new version of mmdetection3d(1.1). When training with a single GPU, I could not find the gpu-id setting in the tools/train.py file. Please give me some information on how to set it up. Thanks.

**Comment by ZCMax** [comment, marked as answer]
You can try just use `CUDA_VISIBLE_DEVICES=1 python train.py xxxx`

**Reply by JiSuanJiDaWang** [reply]
Thanks! I also noticed in the configs/_base_/datset file, the new version variable 'samples_per_gpu' and 'workers_per_gpu'. I wonder what should I do if I want to set up the variable. Just simply change the batch_size and num_workers?

**Reply by ZCMax** [reply]
I do not understand what is your meaning about 'set up the variable'

**Reply by JiSuanJiDaWang** [reply]
I mean if I want to change the number of samples on each GPU or the number of GPU going to be used. Should I just modify batch_size and the num_workers ?

**Reply by ZCMax** [reply]
'samples_per_gpu'  represents the number of samples on each GPU, but if you want to change the number of GPU going to be used, you can use `CUDA_VISIBLE_DEVICES=1, 2, 3, 4...` to change the gpu the program can "see", `workers_per_gpu` does not mean the the number of GPUs going to be used.

**Reply by JiSuanJiDaWang** [reply]
got it. Thanks a lot!
